Source/EasyNetQ.Examples.DlxRetry/EntryPoint.cs

```diff
@@ -14,26 +14,28 @@
 );
 
 await bus.Advanced.QueueDeclareAsync(
-    "Events:Failed",
-    c => c.WithQueueType(QueueType.Quorum)
-        .WithDeadLetterExchange(Exchange.Default)
-        .WithDeadLetterRoutingKey("Events")
-        .WithMessageTtl(TimeSpan.FromSeconds(5)) // A fixed delay between retry attempts
-        .WithOverflowType(OverflowType.RejectPublish)
-        .WithDeadLetterStrategy(DeadLetterStrategy.AtLeastOnce)
+    queue: "Events:Failed",
+    arguments: new Dictionary<string, object>()
+        .WithQueueType(QueueType.Quorum)
```

Comment:
what if introduce QueueArgumets and EchangeArguments  which will be just a proxy for dictionary and could be costracted from dictionary and converted to dictionary but it will be explicit type for args. And also use the same approach with extensions.

```
public class QueueArguments
{
  public QueueArguments() {}
  public QueueArguments(Dictionary initial) {}
  public Dictionary ToDict()
  //add implicit cast from/to dict
}
``` 
or something like this. 
this approach is more strict and you do not need special knowleage about dictionary generic args <string, object> if we are going to settings not as inline method argument value

also accept `QueueArguments`/`ExchangeArguments` types in methods instead of dict but implicit cast will allow to pass dict. but not sure if it is needed to pass dict instead of strict type

Reply:
https://github.com/EasyNetQ/EasyNetQ/pull/1681